### DISCUSSION TITLE
跨服务使用@Echo

### DISCUSSION THREAD

**Discussion by teng-jy** [zuihou/zuihou-admin-cloud#173] [02.提问]
我新服务想用@Echo回显base 服务的组织名称，流程是： 我在base里面建一个feign 继承LoadService 然后重写findByIds 我也要在OrgController中写这个方法 然后将方法一直写到BaseOrgManagerImpl这里，然后我在我新服务里字段上@Echo(api= base里的feignServer)吗？
![%8KH)2AVV~)D RE~XL(F{(4](https://user-images.githubusercontent.com/62694039/186829164-21a2ce59-35b2-4df7-97d8-cf76b9a87ccd.jpg)

->
![@(J9SO%@ N5$)5I@1P$Z$C](https://user-images.githubusercontent.com/62694039/186829173-54532761-3ebf-4b6d-b207-3336c82b2390.jpg)

->

![BZAD(3`{3BYU1O4E_1}4QMU](https://user-images.githubusercontent.com/62694039/186829181-d31fbdd7-d233-44ec-b22d-b76bcfb4b68d.jpg)

->
![OLLI44@{PJ84355@0 KGJ5B](https://user-images.githubusercontent.com/62694039/186829192-b0235139-0078-4daa-8262-411e5eb52c46.jpg)

->
![0L_1HKP1A8%F$_%Q~GXIKKH](https://user-images.githubusercontent.com/62694039/186829235-6febba51-0d72-406f-9a1f-9b1eb5cf6782.jpg)

->
![M0JEMNWL%{3J1}{GB 4CM(7](https://user-images.githubusercontent.com/62694039/186829248-bebb33c1-b5d6-4f5c-9a78-5d3c4de9480e.jpg)

 这是一个大概的流程

**Comment by zuihou** [comment]
新服务里面要加 base-api 的依赖， 并确保新服务里面能正常调用  BaseOrgApi#findByIds

**Reply by teng-jy** [reply]
我在新服务的server模块加入了依赖 也是不可以

**Comment by zuihou** [comment]
手写调用 BaseOrgApi#findByIds

**Comment by zuihou** [comment]
![image](https://user-images.githubusercontent.com/8500410/186842862-0d2c9a3a-0244-43b2-8ac7-de2bdeb3c34d.png)
启动看 strategyMap 里面是否有baseOrgApi， 没有就一定回显失败。


加入以后后，启动时会自动扫描进去。 没进去就说明依赖没加对。

手动能调用这个方法，就说明依赖加对了，strategyMap自然也有了

**Comment by zuihou** [comment]
还有启动类 加 @EnableFeignClients 注解，并指定正确的 basePackages
